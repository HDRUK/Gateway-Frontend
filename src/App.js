import React from "react";

import LandingPage from "./Containers/LandingPage/LandingPage.js";
import AppContext from "./HOC/AppContext/AppContext.js";
import Apollo from "./HOC/Apollo/Apollo.js";

class App extends React.Component {
    render() {
        return (
            <Apollo>
                <AppContext>
                    <LandingPage />
                </AppContext>
            </Apollo>
        );
    }
}

export default App;
