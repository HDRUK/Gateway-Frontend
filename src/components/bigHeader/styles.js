import styled from "styled-components";
import { HeaderWrapper } from "../../styles/carbonComponents.js";

export const BigHeaderWrapper = styled(HeaderWrapper)`
    display: block;
    height: 33.5rem;
    padding-top: 2rem;
`;

export const BigHeaderImage = styled.img`
    height: 6.4375rem;
    display: block;
`;
