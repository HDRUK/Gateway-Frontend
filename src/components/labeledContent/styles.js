import styled from "styled-components";

export const LabeledContentWrapper = styled.div`
    display: inline-block;

    h2,
    p {
        padding-top: 0.125rem;
        padding-right: 0.5rem;
    }
`;

export const HeaderLabel = styled.h2`
    font-size: 0.625rem;
    line-height: 0.75rem;
    color: #979797;
    text-transform: uppercase;
`;

export const ContentText = styled.p`
    font-size: 0.875rem;
    line-height: 1.0625rem;
    color: #0f62fe;

    ${p => !p.lowercase && "text-transform: uppercase"}
`;
