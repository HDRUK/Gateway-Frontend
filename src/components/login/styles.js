import styled from "styled-components";

export const LoginWrapper = styled.span`
    padding: 0 2rem;
    text-align: center;
`;
