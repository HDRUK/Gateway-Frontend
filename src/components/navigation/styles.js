import styled from "styled-components";
import { HeaderNavigation } from "carbon-components-react";

export const NavigationWrapper = styled.div`
    width: 100%;
    height: ${p => p.theme.sizes.navigation}rem;
    border-bottom: 0.0625rem solid ${p => p.theme.colors.border};
`;

export const NavigationHeader = styled(HeaderNavigation)`
    :before {
        content: unset;
    }

    @media (max-width: 66rem) {
        display: block;
    }
`;

export const NavigationItem = styled.div`
    height: 100%;
    min-height: 2.875rem;
    padding: 1rem;
    line-height: 0.875rem;
    text-decoration: none;
    color: ${p => p.theme.text.primary};

    ${p => p.active && `border-bottom: 0.25rem solid ${p.theme.colors.primarySolid}`};
    font-weight: ${p => p.active && p.theme.text.bold};

    :hover,
    :active {
        border-bottom: 0.25rem solid ${p => p.theme.colors.primarySolid};
        font-weight: ${p => p.theme.text.bold};
    }
`;
