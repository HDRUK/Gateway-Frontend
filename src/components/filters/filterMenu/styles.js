import styled from "styled-components";
import { AccordionBlock } from "../../../styles/carbonComponents";

export const ExpandingAccordionBlock = styled(AccordionBlock).attrs(p => ({
    style: {
        paddingBottom: p.expandheight ? `${p.expandheight}px` : "1rem"
    }
}))`
    display: inline-block;
    padding-bottom: "1rem";
`;
