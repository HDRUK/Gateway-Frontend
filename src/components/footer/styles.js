import styled from "styled-components";
import { WidthWrapper, Logo } from "../../styles/styles.js";

export const FooterWrapper = styled.div`
    height: ${p => p.theme.sizes.footer}rem;
    background: linear-gradient(135deg, ${p => p.theme.colors.primary}, ${p => p.theme.colors.secondary});
    color: white;
    padding-top: 4rem;
    padding-bottom: 1rem;
    font-size: 1.1rem;
    position: relative;
`;

export const FooterContent = styled(WidthWrapper)`
    display: block;
    margin: 0 auto 0;
    max-width: 62rem;
    padding: 0 2rem;
`;

export const FooterBlock = styled.div`
    display: inline-block;
    box-sizing: border-box;
    width: 33.33%;
    height: 100%;
    vertical-align: top;
    padding: 0 4rem 0 0;
    line-height: 1.1875rem;

    :last-of-type {
        padding-right: 0;
        padding-left: 4rem;
    }
`;

export const FooterText = styled.div`
    width: 100%;
    margin: 0 0 0.4rem 0;
`;

export const FooterImage = styled(Logo)`
    max-width: 8rem;
    max-height: 5rem;
`;
