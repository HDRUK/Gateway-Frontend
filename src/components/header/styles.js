import styled from "styled-components";
import { LinkNoDecoration } from "../../styles/styles.js";

export const HeaderAlignment = styled.div`
    width: 100%;
`;

export const HeaderAlignmentItem = styled.div`
    height: 100%;
    float: ${p => (p.right ? "right" : "left")};
`;

export const HeaderSpacer = styled.div`
    height: 100%;
    width: 6rem;
`;

export const HeaderLink = styled.a`
    :hover {
        text-decoration: underline;
        text-decoration-color: ${p => p.theme.colors.white};
    }
`;
