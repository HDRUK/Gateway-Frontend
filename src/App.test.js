import React from "react";
import { create } from "react-test-renderer";
import App from "./App";

import { BrowserRouter, Switch, Route } from "react-router-dom";

import Apollo from "./HOC/Apollo/Apollo.js";
import AppContext from "./HOC/AppContext/AppContext.js";
import LandingPage from "./Containers/LandingPage/LandingPage.js";
import Header from "./components/header/header.js";
import Content from "./components/content/content.js";
import { PageWrapper, AppWrapper } from "./styles/styles.js";

import GlobalStyle from "./styles/globalStyles.js";

describe("<App> ", () => {
    let renderedComponent;
    let renderedOutput;

    beforeEach(() => {
        renderedComponent = create(<App />);
        renderedOutput = renderedComponent.root;
    });

    it("should check the correct components are rendered", () => {
        const apollo = renderedOutput.findByType(Apollo);
        const appContext = apollo.findByType(AppContext);
        const appWrapper = appContext.findByType(AppWrapper);
        const browserRouter = appWrapper.findByType(BrowserRouter);
        browserRouter.findByType(GlobalStyle);
        browserRouter.findByType(Header);
        const pageWrapper = browserRouter.findByType(PageWrapper);
        const switches = pageWrapper.findByType(Switch);
        const routes = switches.findAllByType(Route);

        // QUESTION - because of switch only one route renders - how do we test others?

        //first route
        expect(routes[0].props.path).toBe("/");
        expect(routes[0].props.exact).toBe(true);
        const route1Content = routes[0].findByType(Content);
        route1Content.findByType(LandingPage);

        // //second route
        // expect(routes[1].props.path).toBe("/search");
        // const route2Content = routes[1].findByType(Content);
        // route2Content.findByType(SearchPage);

        // //third route
        // expect(routes[2].props.path).toBe("/my-searches");
        // const route3Content = routes[2].findByType(Content);
        // route3div = route3Content.findByType("div");
        // console.log(route3div.children[0]);

        // //fourth route
        // expect(routes[3].props.path).toBe("/browse");
        // const route4Content = routes[3].findByType(Content);
        // route4div = route4Content.findByType("div");

        // //fifth route
        // expect(routes[4].props.path).toBe("/about");
        // const route5Content = routes[4].findByType(Content);
        // route5Content.findByType(AboutPage);

        // //sixth route
        // expect(routes[5].props.path).toBe("/help");
        // const route6Content = routes[5].findByType(Content);
        // route6div = route6Content.findByType("div");
    });
});
