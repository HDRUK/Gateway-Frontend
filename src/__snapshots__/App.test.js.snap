// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App>  should test the App Snapshot 1`] = `
<div
  className="sc-jlyJG cQVlKh"
>
  <header
    className="bx--header sc-eHgmQL snZzi"
    role="banner"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        className="sc-cvbbAY eoOpqv"
        src="hdruk_logo_white.png"
      />
    </a>
    <h1
      className="sc-EHOje sc-jWBwVP hGkMtf"
    >
      Innovation Gateway
    </h1>
  </header>
  <div
    className="sc-gipzik jNbBzy"
  >
    <div
      className="sc-chPdSV iiMNDk"
    >
      <div
        className="sc-jzJRlG sc-cSHVUG jmqzxF"
      />
      <div
        className="sc-jzJRlG sc-kAzzGY eUZgzz"
      />
      <div
        className="sc-dnqmqq IxXZa"
      />
      <div
        className="sc-gzVnrw ddkUmy"
      >
        <text
          className="sc-bdVaJa dcLIZV"
        >
          <h1
            className="sc-EHOje gmsUV"
          >
            Log in to access our datasets
          </h1>
          <div
            className="sc-dnqmqq IxXZa"
          />
          <div
            className="sc-gzVnrw ddkUmy"
          >
            <button
              className="sc-gZMcBi jZLWjr bx--btn bx--btn--primary"
              disabled={false}
              tabIndex={0}
              type="button"
            >
              Log in
            </button>
            <div
              className="sc-htoDjs dPlVZm"
            />
            <a
              href="/search"
              onClick={[Function]}
            >
              <button
                className="sc-gZMcBi jZLWjr bx--btn bx--btn--secondary"
                disabled={false}
                tabIndex={0}
                type="button"
              >
                Continue without logging in
              </button>
            </a>
          </div>
          <div
            className="sc-iwsKbI eSJicE"
          />
        </text>
      </div>
      <text
        className="sc-bdVaJa dcLIZV"
      >
        <h2
          className="sc-bZQynM htZQmN"
        >
          What is the Innovation Gateway
        </h2>
        <div
          className="sc-htoDjs dPlVZm"
        />
        <text
          className="sc-bdVaJa dcLIZV"
        >
          <p
            className="sc-bwzfXH kcFaQD"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
          </p>
        </text>
      </text>
      <div
        className="sc-htoDjs dPlVZm"
      />
      <div
        className="sc-kkGfuU igLqJb"
      >
        <img
          className="sc-iAyFgw KviLY"
          src="hdruk_black.png"
        />
        <img
          className="sc-iAyFgw KviLY"
          src="hdruk_black.png"
        />
        <img
          className="sc-iAyFgw KviLY"
          src="hdruk_black.png"
        />
      </div>
      <div
        className="sc-htoDjs dPlVZm"
      />
      <h2
        className="sc-bZQynM htZQmN"
      >
        HDR News
      </h2>
      <div
        className="sc-htoDjs dPlVZm"
      />
      <div
        className="sc-hMqMXs cPxEnv"
      >
        <div
          className="sc-kEYyzF eLpRwF"
        >
          <div
            className="sc-kpOJdX dozQIn"
          >
            <div
              className="sc-dxgOiQ djCWyI"
              src=""
            />
            <div
              className="sc-ckVGcZ OQpwp"
            >
              <div
                className="sc-jKJlTe jRDNEB"
              >
                HDR release news of new partners in exciting new digital project
              </div>
              <a
                className="sc-eNQAEJ eMaKvB"
                href="http://localhost:3000"
              >
                Read more
              </a>
            </div>
          </div>
        </div>
        <div
          className="sc-kEYyzF eLpRwF"
        >
          <div
            className="sc-kpOJdX dozQIn"
          >
            <div
              className="sc-dxgOiQ djCWyI"
              src=""
            />
            <div
              className="sc-ckVGcZ OQpwp"
            >
              <div
                className="sc-jKJlTe jRDNEB"
              >
                HDR release news of new partners in exciting new digital project
              </div>
              <a
                className="sc-eNQAEJ eMaKvB"
                href="http://localhost:3000"
              >
                Read more
              </a>
            </div>
          </div>
        </div>
        <div
          className="sc-kEYyzF eLpRwF"
        >
          <div
            className="sc-kpOJdX dozQIn"
          >
            <div
              className="sc-dxgOiQ djCWyI"
              src=""
            />
            <div
              className="sc-ckVGcZ OQpwp"
            >
              <div
                className="sc-jKJlTe jRDNEB"
              >
                HDR release news of new partners in exciting new digital project
              </div>
              <a
                className="sc-eNQAEJ eMaKvB"
                href="http://localhost:3000"
              >
                Read more
              </a>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-htoDjs dPlVZm"
      />
      <text
        className="sc-bdVaJa dcLIZV"
      >
        <h2
          className="sc-bZQynM htZQmN"
        >
          Guidelines for using the Innovation Gateway
        </h2>
        <div
          className="sc-htoDjs dPlVZm"
        />
        <text
          className="sc-bdVaJa dcLIZV"
        >
          <p
            className="sc-bwzfXH kcFaQD"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
          </p>
        </text>
        <ul
          className="bx--list--unordered sc-htpNat cGohxE"
        >
          <li
            className="sc-bxivhb lfWzjM"
          >
            Lorem ipsum
          </li>
          <li
            className="sc-bxivhb lfWzjM"
          >
            Lorem ipsum
          </li>
          <li
            className="sc-bxivhb lfWzjM"
          >
            Lorem ipsum
          </li>
        </ul>
        <div
          className="sc-dnqmqq IxXZa"
        />
        <h2
          className="sc-bZQynM htZQmN"
        >
          Help using the Innovation Gateway
        </h2>
        <div
          className="sc-htoDjs dPlVZm"
        />
        <text
          className="sc-bdVaJa dcLIZV"
        >
          <p
            className="sc-bwzfXH kcFaQD"
          >
            There are several ways to get help using the Innovation Gateway
          </p>
        </text>
      </text>
      <ul
        className="bx--list--unordered sc-htpNat cGohxE"
      >
        <li
          className="sc-bxivhb lfWzjM"
        >
          <a
            className="sc-ifAKCX lgyFgY"
          >
            Frequently asked questions
          </a>
        </li>
        <li
          className="sc-bxivhb lfWzjM"
        >
          <a
            className="sc-ifAKCX lgyFgY"
          >
            Forum
          </a>
        </li>
        <li
          className="sc-bxivhb lfWzjM"
        >
          <a
            className="sc-ifAKCX lgyFgY"
          >
            Get in touch with HDR UK
          </a>
        </li>
      </ul>
    </div>
  </div>
  <div
    className="sc-brqgnP bJahzf"
  >
    <div
      className="sc-cMljjf hvrdUf"
    >
      <img
        className="sc-gPEVay EkchX"
        src="hdruk_logo_white.png"
      />
    </div>
    <div
      className="sc-cMljjf hvrdUf"
    >
      <div
        className="sc-jDwBTQ gMXRbf"
      >
        Follow us on social media
      </div>
      <div
        className="sc-iRbamj cUvOZS"
      >
        <a
          href="https://twitter.com/HDR_UK"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            aria-hidden={true}
            focusable="false"
            height={32}
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "willChange": "transform",
              }
            }
            viewBox="0 0 32 32"
            width={32}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M11.92 24.94A12.76 12.76 0 0 0 24.76 12.1v-.59A9.4 9.4 0 0 0 27 9.18a9.31 9.31 0 0 1-2.59.71 4.56 4.56 0 0 0 2-2.5 8.89 8.89 0 0 1-2.86 1.1 4.52 4.52 0 0 0-7.7 4.11 12.79 12.79 0 0 1-9.3-4.71 4.51 4.51 0 0 0 1.4 6 4.47 4.47 0 0 1-2-.56v.05a4.53 4.53 0 0 0 3.55 4.45 4.53 4.53 0 0 1-2 .08A4.51 4.51 0 0 0 11.68 21a9.05 9.05 0 0 1-5.61 2A9.77 9.77 0 0 1 5 22.91a12.77 12.77 0 0 0 6.92 2"
            />
          </svg>
        </a>
      </div>
      <div
        className="sc-iRbamj cUvOZS"
      >
        <a
          href="https://www.linkedin.com/company/healthdataresearchuk/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <svg
            aria-hidden={true}
            focusable="false"
            height={32}
            preserveAspectRatio="xMidYMid meet"
            style={
              Object {
                "willChange": "transform",
              }
            }
            viewBox="0 0 32 32"
            width={32}
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M26.21 4H5.79A1.78 1.78 0 0 0 4 5.73V26.2a1.77 1.77 0 0 0 1.79 1.73h20.42A1.77 1.77 0 0 0 28 26.2V5.73A1.78 1.78 0 0 0 26.21 4zm-15.1 20.41H7.59V13h3.52zm-1.72-13a2.07 2.07 0 0 1-2.07-2.02 2 2 0 0 1 2.07-2.07 2.07 2.07 0 0 1 0 4.13zm15.09 12.93H21v-5.58c0-1.33 0-3.06-1.86-3.06S17 17.16 17 18.63v5.65h-3.56V13h3.32v1.5h.07a3.72 3.72 0 0 1 3.39-1.86c3.59 0 4.26 2.4 4.26 5.45z"
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      className="sc-cMljjf hvrdUf"
    >
      <div
        className="sc-jDwBTQ gMXRbf"
      >
        Visit the HDR UK Site
      </div>
      <div
        className="sc-jDwBTQ gMXRbf"
      >
        Contact us
      </div>
      <div
        className="sc-jDwBTQ gMXRbf"
      >
        Accessibility statement
      </div>
      <div
        className="sc-jDwBTQ gMXRbf"
      >
        Terms and conditions
      </div>
    </div>
    <text
      className="sc-jAaTju ldrpA"
    >
      ©HDR UK 2019. All rights reserved. | Design and development by IBM IX
    </text>
  </div>
</div>
`;
